### PR TITLE
feat: network-statistics api

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- Implement the `/network-statistics` endpoint
  - Uses elasticsearch to query transaction basic statistics such number of transactions, number of blocks and heighest height

### Security Checklist
- [ ] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: serverless.yml
Comment (andreabadesso) [inline serverless.yml:696]: Lambda handles the OPTION request automatically? Does that count as a lambda call?

**Review by andreabadesso — APPROVED** [review]

File: serverless.yml
Comment (alexruzenhack) [reply serverless.yml:696]: Oh! This is a configuration for the API Gateway. But I need to test it after the deploy.

Comment (luislhl) [reply serverless.yml:696]: Is it really needed? I thought this was implied since we are enabling CORS.
 If so, shouldn't it be `OPTIONS`?

Comment (luislhl) [reply serverless.yml:696]: Besides, you can test this in our dev environment before merging this PR. I can show you how.
This way we won't have to create new PRs for fixes.

File: gateways/network_statistics_api_gateway.py
Comment (luislhl) [inline gateways/network_statistics_api_gateway.py:43]: I miss some kind of error treatment. Can't we have error when sending requests to Elasticsearch?

Comment (alexruzenhack) [reply gateways/network_statistics_api_gateway.py:43]: Ok. Added.

File: gateways/network_statistics_api_gateway.py
Comment (alexruzenhack) [reply gateways/network_statistics_api_gateway.py:16]: I have added a smoke load test. I will open a PR to create the Github Action to trigger it manually.

File: .github/workflows/deploy.yml
Comment (luislhl) [inline .github/workflows/deploy.yml:5]: We should undo this before merging

File: .github/workflows/deploy.yml
Comment (luislhl) [inline .github/workflows/deploy.yml:46]: We should undo this before merging

File: tests/load/network_statistics_smoke_test.js
Comment (luislhl) [inline tests/load/network_statistics_smoke_test.js:20]: What is this url? It's the one you get when running the service locally?

How to run it locally?

File: tests/load/network_statistics_smoke_test.js
Comment (luislhl) [inline tests/load/network_statistics_smoke_test.js:23]: This means each user in the test will send 1 request/s, right?

**Review by luislhl — APPROVED** [review]

File: usecases/network_statistics_api.py
Comment (andreabadesso) [inline usecases/network_statistics_api.py:26]: This message will eventually appear in alerts, right?

Maybe add some more details to it, like "ExplorerService failed to fetch statistics from ElasticSearch"

File: tests/load/network_statistics_smoke_test.js
Comment (andreabadesso) [reply tests/load/network_statistics_smoke_test.js:20]: Maybe get the url from args, defaulting to localhost:3002

**Review by andreabadesso — APPROVED** [review]

File: .github/workflows/deploy.yml
Comment (alexruzenhack) [reply .github/workflows/deploy.yml:5]: Done: 3d732c8b5f5671ca12423c63fc69fa0ec1fe050d

File: .github/workflows/deploy.yml
Comment (alexruzenhack) [reply .github/workflows/deploy.yml:46]: Done: 3d732c8b5f5671ca12423c63fc69fa0ec1fe050d

File: tests/load/network_statistics_smoke_test.js
Comment (alexruzenhack) [reply tests/load/network_statistics_smoke_test.js:20]: > How to run it locally?

There is this [`README.md`](https://github.com/HathorNetwork/hathor-explorer-service/blob/e45f58f450f9fa6c75a080ec16fd3cde823677c1/tests/load/README.md) file in the `tests/load` folder.

Comment (alexruzenhack) [reply tests/load/network_statistics_smoke_test.js:20]: > Maybe get the url from args, defaulting to localhost:3002

Done: 5bae2b0b21b2a7d7fe09ca23dd009b61c0a4b881

File: tests/load/network_statistics_smoke_test.js
Comment (alexruzenhack) [reply tests/load/network_statistics_smoke_test.js:23]: Right!

File: usecases/network_statistics_api.py
Comment (alexruzenhack) [reply usecases/network_statistics_api.py:26]: Done: 654d151b516001b6086e273e148e2b2e8c3964e4

File: tests/load/network_statistics_smoke_test.js
Comment (tuliomir) [reply tests/load/network_statistics_smoke_test.js:23]: I would suggest this information to be in a comment.
These default units passed as parameters can be a source of confusion to the reader/maintainer.

File: usecases/network_statistics_api.py
Comment (tuliomir) [inline usecases/network_statistics_api.py:32]: Wouldn't it be better to raise the original error message after writing to the logger here?

I feel we can be a bit lost debugging whenever these `TransportError`s occur, because their message has been suppressed.

**Review by tuliomir — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

File: usecases/network_statistics_api.py
Comment (alexruzenhack) [reply usecases/network_statistics_api.py:32]: We can't raise the original error because it would expose internal information to the client. We can add the error to the error message in other PR, indeed.G

**Pull request merged**
